src/BusinessChat/BCChatAction.cs

```diff
@@ -0,0 +1,15 @@
+#if XAMCORE_2_0
+using System;
+using System.Collections.Generic;
+using XamCore.Foundation;
+
+namespace XamCore.BusinessChat {
+	public partial class BCChatAction {
+		public static void OpenTranscript (string businessIdentifier, Dictionary<BCParameterName, string> intentParameters) {
+			var keys = Array.ConvertAll<string, NSString> (new List<string>(intentParameters.Values).ToArray (), v => new NSString (v));
```

Comment:
that's creating a **lot** of short-lived, managed objects (e.g. `ToArray`)

Reply:
in particular `new List<string>(intentParameters.Values).ToArray ()` is bad
why create a short-lived list if what you want is an array